### PR TITLE
minimal fees

### PR DESCRIPTION
**Pull request by modship**
* [x] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

Ownership: Massa Foundation

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -106,8 +106,8 @@ massa_versioning = { path = "./massa-versioning" }
 massa_wallet = { path = "./massa-wallet" }
 
 # Massa projects dependencies
-massa-proto-rs = { git = "https://github.com/massalabs/massa-proto-rs", "rev" = "25638b3b7d387afbca81afcb02bae1af03697f18" }
-massa-sc-runtime = { git = "https://github.com/massalabs/massa-sc-runtime", "rev" = "32102813c05559c17d8ac1e1ee673802c2ba2ea8" }
+massa-proto-rs = { git = "https://github.com/massalabs/massa-proto-rs", "rev" = "426fd325a55dfcc4033920bed2de075a7e7ad4b7" }
+massa-sc-runtime = { git = "https://github.com/massalabs/massa-sc-runtime", "rev" = "5a858d7e34c9f88c047588a31d816cab98ad8f1b" }
```
Comment (modship) [inline Cargo.toml:110]: this isn"t merge : https://github.com/massalabs/massa-sc-runtime/tree/update_proto_minimal_fee

**Review by AurelienFT — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (AurelienFT) [reply Cargo.toml:110]: It is now ! :)

File: massa-pool-worker/src/operation_pool.rs [hunk 2]
```diff
@@ -494,6 +494,13 @@ impl OperationPool {
                 continue;
             }
 
+            // if minimum fees are required, check that the fee is enough
+            if let Some(minimal_fees) = self.config.minimal_fees {
```
[the remark this answers is not in the document]
Comment (modship) [reply massa-pool-worker/src/operation_pool.rs]: moved

File: massa-api/src/public.rs [hunk 3]
```diff
@@ -281,6 +281,20 @@ impl MassaRpcServer for API<Public> {
                 fee,
             };
 
+            if let Some(minimal_fees) = self.0.api_settings.minimal_fees {
+                if fee.map_or(true, |fee| fee.checked_sub(minimal_fees).is_none()) {
+                    let result = ExecuteReadOnlyResponse {
+                        executed_at: Slot::new(0, 0),
+                        result: ReadOnlyResult::Error("fee is too low".to_string()),
```
[the remark this answers is not in the document]
Comment (modship) [reply massa-api/src/public.rs]: added

File: massa-api/src/public.rs [hunk 4]
```diff
@@ -281,6 +281,20 @@ impl MassaRpcServer for API<Public> {
                 fee,
             };
 
+            if let Some(minimal_fees) = self.0.api_settings.minimal_fees {
+                if fee.map_or(true, |fee| fee.checked_sub(minimal_fees).is_none()) {
```
[the remark this answers is not in the document]
Comment (modship) [reply massa-api/src/public.rs]: should be more readable now

**Review by AurelienFT — APPROVED** [review]

**Pull request merged**